src/lpython/semantics/python_intrinsic_eval.h

```diff
@@ -77,20 +77,34 @@ struct IntrinsicNodeHandler {
                 char *c = ASR::down_cast<ASR::StringConstant_t>(
                                     ASRUtils::expr_value(arg))->m_s;
                 int ival = 0;
+                std::string str;
                 char *ch = c;
-                if (*ch == '-') {
+                if (*ch == '-' || *ch == '+') {
                     ch++;
                 }
                 while (*ch) {
+                    if(*ch == ' '){
+                        ch++;
+                        continue;
+                    }
+                    if(*ch == '\\'){
+                        ch++;
+                        if (*ch == 'n' || *ch == 'r' || *ch == 't') {
+                            ch++;
+                            continue;
+                        }
+                        throw SemanticError("invalid literal for int() with base 10: '"+ std::string(c) + "'", arg->base.loc);
+                    }
                     if (*ch == '.') {
                         throw SemanticError("invalid literal for int() with base 10: '"+ std::string(c) + "'", arg->base.loc);
                     }
                     if (*ch < '0' || *ch > '9') {
                         throw SemanticError("invalid literal for int() with base 10: '"+ std::string(c) + "'", arg->base.loc);
                     }
+                    str+=*ch;
                     ch++;
                 }
-                ival = std::stoi(c);
+                ival = std::stoi(str);
```

Comment:
Edge case to be looked into :
```
def f():
    print(int(' '))    # Only whitespace without any numeric character.
f()

(lp) C:\Users\kunni\lpython>python try.py
Traceback (most recent call last):
  File "C:\Users\kunni\lpython\try.py", line 3, in <module>
    f()
  File "C:\Users\kunni\lpython\try.py", line 2, in f
    print(int(' '))
ValueError: invalid literal for int() with base 10: ' '

(lp) C:\Users\kunni\lpython>src\bin\lpython try.py
std::exception: invalid stoi argument
```

Been a bit busy with university exams, will handle this soon.

Reply:
Does this case throw an error?

Reply:
Yes this case throws an error - The `stoi` catches invalid input and throws an error. I'll have to add a if condition and throws a corresponding semantic error. Thats it right?

Reply:
Yes, I think, we have to check and throw a SemanticError.

Reply:
Other than this edge case I feel all cases are handled !